compiler/AST/expr.cpp

```diff
@@ -476,9 +476,17 @@ GenRet SymExpr::codegen() {
   } else {
 #ifdef HAVE_LLVM
     if(isVarSymbol(var)) {
-      ret = toVarSymbol(var)->codegen();
+      VarSymbol* varSym = toVarSymbol(var);
+      ret = varSym->codegen();
+      if( varSym->hasFlag(FLAG_CONST) ) {
+        ret.baseTypeConstant = 1;
+      }
     } else if(isArgSymbol(var)) {
-      ret = info->lvt->getValue(var->cname);
+      ArgSymbol* argSym = toArgSymbol(var);
+      ret = info->lvt->getValue(argSym->cname);
+      if( (argSym->intent & INTENT_FLAG_CONST) != 0 ) {
```

Comment:
Does it make sense to make this a method on ArgSymbol? E.g. ArgSymbol::hasConstIntent or something like that ?